entity-framework/core/what-is-new/ef-core-10.0/whatsnew.md

```diff
@@ -21,6 +21,158 @@ EF10 requires the .NET 10 SDK to build and requires the .NET 10 runtime to run.
+            b.Property(x => x.ContentsGerman).EnableFullTextSearch();
+            b.HasIndex(x => x.ContentsGerman).IsFullTextIndex();
+            b.Property(x => x.TagsGerman).EnableFullTextSearch();
+            b.HasIndex(x => x.TagsGerman).IsFullTextIndex();
+        });
+    }
+```
+
+<a name="full-text-search-querying"></a>
+
+#### Querying
+
+As part of the full-text search feature, Azure Cosmos DB introduced several built-in functions which allow for efficient querying of content inside the full-text search enabled properties. These functions are: `FullTextContains`, `FullTextContainsAll`, `FullTextContainsAny`, which look for specific keyword or keywords and `FullTextScore`, which returns [BM25 score](https://en.wikipedia.org/wiki/Okapi_BM25) based on provided keywords.
+
+> [!NOTE]
+> `FullTextScore` can only be used inside `OrderBy` to rank the documents based on the score.
+
+EF Core exposes these functions as part of `EF.Functions` so they can be used in queries:
+
+```c#
+var cosmosBlogs = await context.Blogs.Where(x => EF.Functions.FullTextContainsAll(x.Contents, "database", "cosmos")).ToListAsync();
+
+var keywords = new string[] { "AI", "agent", "breakthrough" };
+var mostInteresting = await context.Blogs.OrderBy(x => EF.Functions.FullTextScore(x.Contents, keywords)).Take(5).ToListAsync();
+```
+
+<a name="full-text-search-hybrid-search"></a>
+
+#### Hybrid search
+
+Full-text search can be used with vector search in the same query (i.e. hybrid search), by combining results of `FullTextScore` and `VectorDistance` functions. It can be done using the `RRF` function (Reciprocal Rank Fusion), which EF Core also provides inside `EF.Functions`:
+
+```c#
+public class Blog
+{
+    ...
+
+    public float[] Vector { get; set; }
+    public string Contents { get; set; }
+}
+
+public class BloggingContext
+{
+    ...
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.Entity<Blog>(b =>
+        {
+            b.Property(x => x.Contents).EnableFullTextSearch();
+            b.HasIndex(x => x.Contents).IsFullTextIndex();
+
+            b.Property(x => x.Vector).IsVectorProperty(DistanceFunction.Cosine, dimensions: 1536);
+            b.HasIndex(x => x.Vector).IsVectorIndex(VectorIndexType.Flat);
+        });
+    }
+}
+
+float[] myVector = ...
+var hybrid = await context.Blogs.OrderBy(x => EF.Functions.Rrf(
+        EF.Functions.FullTextScore(x.Contents, "database"), 
+        EF.Functions.VectorDistance(x.Vector, myVector)))
+    .Take(10)
+    .ToListAsync();
+```
+
+> [!TIP]
+> You can combine more than two scoring functions inside `Rrf` call, as well as using only `FullTextScore`, or only `VectorDistance`.
+
+<a name="full-text-search-limitations"></a>
+
+#### Limitations
+
+Here are some current limitations of the full-text search using EF Core:
+
+- `FullTextScore` and `RRF` functions can only be used inside `OrderBy` and are mutually exclusive with other forms of ordering.
```

Comment:
Are these "current limitations" in EF? Isn't this just how Cosmos works (and not a limitation per se)?

Reply:
RRF in OrderBy is Cosmos, not supporting FTS on collections is EF - you can have FTS property anywhere in the document without issue, but wildcards are not supported (when defining paths for index), so people need to define a path for each element in the collection `/Collection/0/MyFts`, `/Collection/1/MyFts` and so on. This is unrealistic for us to do on EF, so I disabled the feature (so that people don't fall into the pit of using FTS without index)